1.5/Source/PawnStorages/PawnStorages/Farm/Comps/CompFarmBreeder.cs

```diff
@@ -38,13 +90,62 @@ public override void CompTick()
 
             if (parent.IsHashIntervalTick(Parent.TickInterval))
             {
-                if (Parent.BreedablePawns.Count < PawnStoragesMod.settings.MaxPawnsInFarm)
+                var types = (from p in Parent.AllPawns
+                    group p by p.kindDef
+                    into def
+                    select def).ToList();
+
+                if (parent.IsHashIntervalTick(Parent.TickInterval * 4))
                 {
-                    var types = from p in Parent.BreedablePawns
-                        group p by p.kindDef
-                        into def
-                        select def;
 
+                    foreach (var type in types)
+                    {
+                        AutoSlaughterConfig config = AutoSlaughterSettings.FirstOrDefault(s => s.animal == type.Key.race);
+
+                        if (config != null)
+                        {
+                            // Only slaughter one per cycle
```

Comment:
might be overkill but this is so duplicatey I'd consider extracting it into a CullOldest function then you only need to pass in the lists or even just the whole types thing. As you're only culling one you don't need to pre-make all of them either just one at a time; maybe have a culledThisCycle bool then it's just something like

```csharp
bool culledThisCycle = false;
if (!culledThisCycle) culledThisCycle = cullOldestOverLimit(config.maxMales, type.Where(p => p.gender == Gender.Male && p.ageTracker.Adult).ToList());
if (!culledThisCycle) culledThisCycle = cullOldestOverLimit(config.maxFemales, type.Where(p => p.gender == Gender.Female && p.ageTracker.Adult).ToList());
if (!culledThisCycle) culledThisCycle = cullOldestOverLimit(config.maxMales, type.Where(p => p.gender == Gender.Male && !p.ageTracker.Adult).ToList());
if (!culledThisCycle) culledThisCycle = cullOldestOverLimit(config.maxFemales, type.Where(p => p.gender == Gender.Female && !p.ageTracker.Adult).ToList());
```

your CountForType util even already basically does this right

If you really wanted to go ham you could even roll it all into one iteration, just groupBy ageTracker and gender. Then a Find using the cull function which returns true if culled so it'll just do one iteration then cull one from one group and stop. Might be a bit harder to read though so your call.

In general there's way too much going on in this function now too, we're better off extracting the steps to like TryCull, TryBreed etc

Reply:
Completely refactored based roughly on what you suggested. Likely needs a good looking over.